deployments/local-external-minio/deploy.sh

```diff
@@ -0,0 +1,121 @@
+    printf "." >&2
+  done
+}
+
+# TODO Create bucket in Minio if it does not exist yet.
+
+# Stop running stack
+function rm-stack () {
+    printf "Removing $stackName docker stack...\n"
+    docker stack rm $stackName
+    printf "Done!\n"
+}
+
+# Docker networks usually take a few seconds to go down, so it waits to prevent redeployment issues
+function rm-network () {
+    printf "Removing $networkName docker network...\n"
+    docker network rm $networkName
+    printf "\nWaiting for docker network '$networkName' to go down.\n"
+    wait-for-empty-results "docker network ls | grep -w $networkName"
+    printf "\nDone!\n"
+}
+
+# Waits for everything to start up
+function wait-for-containers () {
+    printf "\nWaiting for docker services to start (safe to exit)."
+    wait-for-empty-results "docker service ls | grep $stackName_*0/[1-9]"
+    printf "Done!\n"
+}
+
+# Deploy docker-compose.yml file via docker stacks
+function deploy-images () {
+    printf "\nDeploying Docker Images...\n"
+    docker stack deploy -c ../../docker-compose.yml -c docker-override.yml $stackName
+}
+
+# Create attachable overlay docker network
+function create-network () {
+    printf "\nCreating docker '$networkName' network...\n"
+    docker network create --driver=overlay --attachable $networkName
+}
+
+# Check is network already does not exist or is not attachable.
+function configure-network () {
+    printf "Checking Docker Network...\n"
+    network=`docker network ls | grep "$networkName"`
+    networktype=$(docker network inspect $networkName 2>/dev/null | grep -i "\"driver\": \"overlay\"")
+    attachable=$(docker network inspect $networkName 2>/dev/null | grep -i "\"attachable\": true")
+    if [ -z "$network" ]; then
+        printf "Network '$networkName' does not exist, creating new network... \n"
+        create-network
+    elif [ -z "$networktype" ] || [ -z "$attachable" ]; then
+        printf  "ERROR: The docker network '$networkName' exists, but is not an attachable overlay network. Please run 'clean' or 'redeploy' to remove or update this network.\n"
+        exit 1
+    else
+        printf "Network '$networkName' already exists and is attachable. \n"
+    fi
+}
+
+# Picks up changes to Dockerfiles and Compose files
+function redeploy () {
+    print_warning
+    rm-stack
+    configure-network
+    deploy-images
+    wait-for-containers
+}
+
+# Fastest option. Does not pick up all changes.
+function update () {
+    print_warning
+    configure-network
+    deploy-images
+    wait-for-containers
+}
+
+function print_warning () {
+    printf "====== WARNING: TEST DEPLOYMENT ======\nMake sure this isn't being deployed into production.\n\n"
+}
+
+# start of script
+if [ -z $1 ]; then
+    update
+else
+   case $1 in
+    update | u)
+        update
+        ;;
+    redeploy | r)
+        redeploy
+        ;;
+    clean | c)
+        rm-stack
+        rm-network
+        ;;
+    *)
+        printf "$helptext"
+        ;;
+    esac
+fi
```

Comment:
I noticed there more and more duplication between repositories. It is tech debt :money_with_wings: and I don't have a clever solution for it. I think a good time to tackle the duplicated stuff would be when migrating to a new source control system.